includes/gateways/class-api-gateway.php

```diff
@@ -171,10 +171,15 @@ public function unarchive_poll( $id_to_unarchive ) {
 	 * @return Nps_Survey|WP_Error
 	 */
 	public function update_nps( Nps_Survey $survey ) {
+		$survey_array = $survey->to_array();
+
+		// Inject source attribute.
+		$survey_array['source'] = 'crowdsignal-forms';
```

Comment:
I wonder if this should maybe be part of the `NPS_Survey` class?
It's a constant value so we could inject it as inside the `to_array()` method.

I believe we'll need to add a `source_link` property on that class as well to be compatible with d56246-code?

Reply:
> I wonder if this should maybe be part of the NPS_Survey class?

Wonder as well. While `source` and `source_link` are accepted payload props, they are not required and, actually, not customizable from the client side. I consider these props *hard coded* for a reason: we don't want client side messing with those other than through our automated way of setting them.

Having said that, I placed the `source` injection out of reach of any method, but I'm open to discuss better approaches for it.

> I believe we'll need to add a source_link property on that class as well to be compatible with d56246-code?

As talked earlier, knowing this will require some class|model modification, the changes on d56246-code are only paving the road so the props are handled properly. Next PR will/should have the needed changes/additions to send the props

Reply:
Well, thinking about it some more... this is very much the 'end of the line' before the request go out so it makes some sense in that way. Let's stick with it for now then.